ui_testing/testcases/basic_tests/test08_contacts.py

```diff
@@ -0,0 +1,382 @@
+        count_of_contact_person_after_delete = self.contact_page.get_contact_persons_count()
+
+        if count_of_contact_person_after_delete == count_of_contact_person_before_delete:
+            is_successfully_deleted = self.contact_page.check_contact_persons_table_is_empty()
+            if is_successfully_deleted:
+                self.base_selenium.LOGGER.info('Contact person removed successfully')
+            else:
+                self.base_selenium.LOGGER.info('Contact person was not removed successfully, a bug should be created')
+                self.assertEqual(True, is_successfully_deleted)
+        else:
+            self.base_selenium.LOGGER.info('Contact person removed successfully')
+        
+        self.base_selenium.LOGGER.info('getting contact person data after remove the record to check that it was not corrupted after being saved')
+        contact_persons_after_remove = self.contact_page.get_contact_persons_data()
+
+        self.contact_page.save(save_btn='contact:save')
+        
+        self.base_selenium.LOGGER.info('Refreshing to make sure that data are saved without being corrupted')
+        self.base_selenium.refresh()
+
+        contact_data_after_refresh = self.contact_page.get_full_contact_data()
+        if self.contact_page.compare_contact_main_data(data_after_save=contact_data_after_refresh, data_before_save=contact_data):
+            self.base_selenium.LOGGER.info('contact data have been saved successfully')
+        else:
+            self.base_selenium.LOGGER.info('contact data was not saved successfully, you should report a BUG')
+            self.assertEqual(True, False)
+
+        self.contact_page.get_contact_persons_page()
+        self.base_selenium.LOGGER.info('get contact person data after refresh to make sure it was saved correctly')
+        
+        contact_persons_after_refresh = self.contact_page.get_contact_persons_data()
+        if count_of_contact_person_before_delete == 1:
+            if self.contact_page.check_contact_persons_table_is_empty():
+                self.base_selenium.LOGGER.info('contact person deleted successfully')
+            else:
+                self.base_selenium.LOGGER.info('contact person was not deleted successfully, report a bug')
+                self.assertEqual(True, False)
+
+        if len(contact_persons_after_refresh) == len(contact_persons_after_remove):
+            if self.contact_page.compare_contact_persons_data(data_after_save=contact_persons_after_refresh, data_before_save=contact_persons_after_remove):
+                self.base_selenium.LOGGER.info('contact persons have been saved successfully')
+            else:
+                self.base_selenium.LOGGER.info('contact persons was not saved successfully, you should report a BUG')
+                self.assertEqual(True, False)
+        else:
+            self.base_selenium.LOGGER.info('Data were not correctly saved, report a bug')
+            self.assertEqual(True, False)
```

Comment:
I cant follow here. We get a random contact, delete it, search for it, check there is no results. That is it. No?

Reply:
yea, but it is long as i needed to check on data before pressing save and after pressing save, and had to make sure that case the first record had already more than 2 persons, not just one, in this case i need to check on count and on empty table.